src/test/java/com/google/sps/servlets/UserCardsServletTest.java

```diff
@@ -0,0 +1,199 @@
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.gson.Gson;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.data.Folder;
+import com.google.sps.data.Card;
+import com.google.sps.tool.EntityTestingTool;
+import java.util.List; 
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+
+@RunWith(JUnit4.class)
+public final class UserCardsServletTest {
+
+  private final LocalServiceTestHelper helper = 
+    new LocalServiceTestHelper(
+      new LocalDatastoreServiceTestConfig()
+        .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+      new LocalUserServiceTestConfig())
+      .setEnvIsAdmin(true).setEnvIsLoggedIn(true)
+      .setEnvEmail("test@gmail.com").setEnvAuthDomain("gmail.com");
+  
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private UserCardsServlet servlet;
+  private DatastoreService datastore;
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+    servlet = new UserCardsServlet();
+    mockRequest = mock(HttpServletRequest.class);
+    mockResponse = mock(HttpServletResponse.class);
+    
+    // Set up a fake HTTP response 
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    // Initialize datastore
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    helper.setEnvIsLoggedIn(true);
+    helper.tearDown();
+  }
+
+  @Test
+  public void QueryUserCards() throws Exception {
+    // Returns array of Cards and signals front-end to show Create Card Option
+
+    // Generate testing card objects to store in datastore
+    Card cardA = new Card("null", "viet", "en", "vi", "test", "test");
+    Card cardB = new Card("null", "viet", "en", "vi", "test", "test");
+    
+    // Generate a dummy Folder Entity
+    Entity folder = new Entity("Folder", "testID");
+    String folderKey = KeyFactory.keyToString(folder.getKey());
+
+    when(mockRequest.getParameter("folderKey")).thenReturn(folderKey);
+    
+    List<Card> cards = new ArrayList<>();
+    Card cardAInDatastore = EntityTestingTool.populateDatastoreWithACard(cardA, datastore, folderKey);
+    Card cardBInDatastore = EntityTestingTool.populateDatastoreWithACard(cardB, datastore, folderKey);
+    cards.add(cardAInDatastore);
+    cards.add(cardBInDatastore);
+
+    servlet.doGet(mockRequest, mockResponse);
+    String response = responseWriter.toString();
+    String expectedResponse = new Gson().toJson(EntityTestingTool.getExpectedJsonCardInfo(
+        /*card*/cards, /*showCreateFormStatus*/true));
+
+    assertTrue(compareJson(response, expectedResponse));
+  }
+
+  @Test
+  public void UserHasNoCurrentCard() throws Exception {
+    // Returns an empty array list and signals front-end to show Create Card Option
+
```

Comment:
Remove blank lines at the beginning and end of a function.

If you want to comment what a function does, you should use a JavaDoc, which goes above the function.